src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2239,13 +2244,17 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
         ASR::symbol_t* v_sym = ASR::down_cast<ASR::symbol_t>(v);
         ASR::Variable_t* v_variable = ASR::down_cast<ASR::Variable_t>(v_sym);
 
-        if( init_expr && current_body &&
+        if( init_expr && (current_body || ASR::is_a<ASR::List_t>(*type)) &&
             (is_runtime_expression || !is_variable_const)) {
             ASR::expr_t* v_expr = ASRUtils::EXPR(ASR::make_Var_t(al, loc, v_sym));
             cast_helper(v_expr, init_expr, true);
             ASR::asr_t* assign = ASR::make_Assignment_t(al, loc, v_expr,
                                                         init_expr, nullptr);
-            current_body->push_back(al, ASRUtils::STMT(assign));
+            if (current_body) {
+                current_body->push_back(al, ASRUtils::STMT(assign));
+            } else if (ASR::is_a<ASR::List_t>(*type)) {
+                global_init.push_back(al, assign);
+            }
```

Comment:
I assumed `current_body` contains `true` value only when `create_add_variable_to_scope` (this function) is visited inside the `Function` or `Program` nodes. In the module or global scope, `current_body` would be `nullptr`.
So, I have added a check for `global_lists`.
Have I handled this correctly? @czgdp1807

Reply:
Make sense to me.